### PR TITLE
✨ #22160 Add new EQUALS macro

### PR DESCRIPTION
**Pull request by jonathantyng-amp**
Add EQUALS macro to allow for easier string output mapping

Resolves issue #22160

### CODE REVIEW

**Comment by jonathantyng-amp** [comment]
Tested manually, ready to be merged!

**Comment by zhouyx** [comment]
Woohoo!

**Comment by zhouyx** [comment]
So @calebcordry and I found an issue with the current nested macro impl, that will affect this PR. Please hold until we have a fix there and then decide what to do with boolean macro argument.

**Comment by zhouyx** [comment]
@jonathantyng-amp The PR is unblocked now. 
As agreed, let's change the `$IF` logic, so it converts string true/false to boolean true/false. Thanks.

File: extensions/amp-analytics/0.1/variables.js [hunk 1]
```diff
@@ -351,3 +355,19 @@ export function getConsentStateStr(element) {
...
   });
 }
+
+/**
+ * Converts string to boolean
+ * @param {string} str
+ * @return {bool}
+ */
+function stringToBool_(str) {
+  console.log(str);
```
Comment (calebcordry) [inline extensions/amp-analytics/0.1/variables.js]: oops :)

Comment (jonathantyng-amp) [reply extensions/amp-analytics/0.1/variables.js]: good catch :)

**Comment by jonathantyng-amp** [comment]
cc @zhouyx ready for merge!

File: extensions/amp-analytics/0.1/variables.js [hunk 2]
```diff
@@ -351,3 +355,18 @@ export function getConsentStateStr(element) {
...
 }
+
+/**
+ * Converts string to boolean
+ * @param {string} str
+ * @return {boolean}
+ */
+function stringToBool_(str) {
+  return (
+    str !== 'false' &&
```
Comment (zhouyx) [inline extensions/amp-analytics/0.1/variables.js]: nit: No need to add the trailing `_` to local function. And can we please also use this helper method here? Thanks 
https://github.com/ampproject/amphtml/blob/2250345ce3e6d2ab903538561f0bd6124b542001/extensions/amp-analytics/0.1/amp-analytics.js#L727-L734

**Comment by zhouyx** [comment]
@jonathantyng-amp is this ready to be merged?

**Comment by jonathantyng-amp** [comment]
@zhouyx yes, ready to be merged!

**Pull request merged**